src/install-pnpm/run.ts

```diff
@@ -29,17 +31,24 @@ export async function runSelfInstaller(inputs: Inputs): Promise<number> {
     cp.on('close', resolve)
   })
   if (exitCode === 0) {
-    const pnpmHome = path.join(dest, 'node_modules/.bin')
+    const pnpmHome = join(dest, 'node_modules/.bin')
     addPath(pnpmHome)
     exportVariable('PNPM_HOME', pnpmHome)
   }
   return exitCode
 }
 
-async function readTarget(packageJsonPath: string, version?: string | undefined) {
+async function readTarget(version?: string | undefined) {
   if (version) return `pnpm@${version}`
 
-  const { packageManager } = JSON.parse(await readFile(packageJsonPath, 'utf8'))
+  const workspace = process.env.GITHUB_WORKSPACE
+  if (!workspace) {
+    throw new Error(`No workspace is found.
+If you're intended to let this action read pnpm version from the package.json/packageManager field,
+please run the actions/checkout before this one, otherwise please specify the version in the action config.`)
```

Comment:
Can you change `package.json/packageManager` to be more readable? Example: "the `"packageManager"` field in the `package.json` file".